test/solc/CommandLineParser.cpp

```diff
@@ -422,6 +420,32 @@ BOOST_AUTO_TEST_CASE(standard_json_mode_options)
 	BOOST_TEST(parsedOptions.value() == expectedOptions);
 }
 
+BOOST_AUTO_TEST_CASE(experimental_via_ir_invalid_input_modes)
+{
+	vector<string> commandLine = {
+		"solc",
+		"--experimental-via-ir",
+		"file",
+	};
+
+	static vector<vector<string>> const allowedCombinations = {
+		{"--assemble"},
+		{"--yul"},
+		{"--strict-assembly"},
+		{"--standard-json"},
+		{"--link"}
```

Comment:
This could just be a flat `vector`. You don't need a vector of vectors. In fact, it would be better to just an `array` since the length is known:
```suggestion
	static array<string, 5> const inputModeOptions = {
		"--assemble",
		"--yul",
		"--strict-assembly",
		"--standard-json",
		"--link",
```